src/deterministic.jl

```diff
@@ -128,3 +126,14 @@ function split_time_series(
 
     return T(vals...)
 end
+
+function make_time_series(info::TimeSeriesParsedInfo{Deterministic})
```

Comment:
I agree that is more similar, but it was weird to me to have this pattern: 

```Julia
struct TimeSeriesInfo
    ts_type::DataType
    other_data::Dict
end

function make_time_series(val::TimeSeriesInfo
    return val.ts_type(val.other)
end
```

I think this is very strange. We call make to return a type we evaluate inside of another one.

Reply:
Here is an idea.  The time series type doesn't need to be part of TimeSeriesParseInfo because you added it to TimeSeriesFileMetadata.  You could change this function in system_data.jl line 225 to

```
function make_time_series!(cache::TimeSeriesCache, ts_file_metadata::TimeSeriesFileMetadata)
    info = add_time_series_info!(cache, ts_file_metadata)
    return ts_file_metadata.time_series_type(info)
end

```

Reply:
I don't see how this is beneficial if we still need to implement the constructors based on info. I think the parametric version is easier to interpret.

Reply:
1. The constructors in single_time_series.jsl and deterministic.jl will follow the same pattern.  Currently, you have the function make_time_series that behaves like a constructor but isn't named like one.
2. The type is already part of TimeSeriesFileMetadata, to which you already have access.  There isn't any value in duplicating it in TimeSeriesParsedInfo.
3. IMHO this isn't how people typically use parametric types.

Reply:
I think that having a function like `make_time_series` would be appropriate if it operated on an abstract type, obviating the need to add a constructor for each concrete type.  But that isn't what you've done.